### PR TITLE
Add `lxc file create` subcommand (from Incus)

### PR DESCRIPTION
**Pull request by kadinsayani**
Cherry-picked from https://github.com/lxc/incus/pull/408. I've also added instance shell completions for `lxc file` subcommands.

Signed-off-by: Kadin Sayani <kadin.sayani@canonical.com>

### CODE REVIEW

File: lxc/file.go [hunk 1]
```diff
@@ -110,6 +118,187 @@ func (c *cmdFile) command() *cobra.Command {
...
+	} else if c.flagType == "file" {
+		mode = os.FileMode(FileMode)
+	}
+
+	if c.file.flagMode != "" {
+		if len(c.file.flagMode) == 3 {
+			c.file.flagMode = "0" + c.file.flagMode
+		}
+
+		m, err := strconv.ParseInt(c.file.flagMode, 0, 32)
```
Comment (simondeziel) [inline lxc/file.go:237]: `os.FileMode` being a `uint32` (https://pkg.go.dev/os#FileMode => https://pkg.go.dev/io/fs#FileMode). Would it make sense to use `strconv.ParseUint` instead?

**Review by simondeziel — APPROVED** [review]
LGTM with a small question

File: lxc/file.go (hunk 1 above)
Comment (kadinsayani) [reply lxc/file.go]: Good catch, I've updated both references in `file.go` and also set a base and upper bound.

**Review by simondeziel — DISMISSED** [review]

**Comment by tomponline** [comment]
tests are unhappy

**Comment by kadinsayani** [comment]
> tests are unhappy

Should be good to go now 👍

**Comment by kadinsayani** [comment]
> tests are unhappy

Should be good to go now 👍

File: lxc/file.go [hunk 2]
```diff
@@ -532,7 +721,7 @@ func (c *cmdFilePush) run(cmd *cobra.Command, args []string) error {
 			c.file.flagMode = "0" + c.file.flagMode
 		}
 
-		m, err := strconv.ParseInt(c.file.flagMode, 0, 0)
+		m, err := strconv.ParseUint(c.file.flagMode, 8, 32)
```
Comment (tomponline) [inline lxc/file.go:756]: This should be a separate commit as not related to new command
